internal/pkg/testproject/project.go

```diff
@@ -222,36 +236,98 @@ func (p *Project) SetState(stateFilePath string) {
 				OnSuccess(func(response *client.Response) {
 					p.logf(`crated branch "%s", id: "%d"`, branch.Name, branch.Id)
 					p.setEnv(fmt.Sprintf("TEST_BRANCH_%s_ID", branch.Name), cast.ToString(branch.Id))
+					p.branchesById[branch.Id] = branch
+					p.branchesByName[branch.Name] = branch
 				}).
 				Send()
 		}
 	}
+}
+
+func (p *Project) createConfigsInDefaultBranch(names []string) {
+	p.branchesById[p.defaultBranch.Id] = p.defaultBranch
+	p.branchesByName[p.defaultBranch.Name] = p.defaultBranch
+
+	// Prepare configs
+	tickets := p.api.NewTicketProvider()
+	configs := p.prepareConfigs(names, p.defaultBranch, tickets,"TEST_BRANCH_ALL_CONFIG")
+
+	// Generate new IDs
+	if err := tickets.Resolve(); err != nil {
+		assert.FailNow(p.t, fmt.Sprintf(`cannot generate new IDs: %s`, err))
+	}
 
-	// Create configs in dev-branches
-	pool = p.api.NewPool()
-	for _, branch := range stateFile.Branches {
-		modelBranch := branchesByName[branch.Branch.Name]
+	// Create requests
+	pool := p.api.NewPool()
+	p.createConfigsRequests(configs, pool)
+
+	// Send requests
+	if err := pool.StartAndWait(); err != nil {
+		assert.FailNow(p.t, fmt.Sprintf("cannot create configs in default branch: %s", err))
+	}
+}
+
+func (p *Project) createConfigs(branches []*fixtures.BranchState, additionalEnvs map[string]string) {
+	// Prepare configs
+	tickets := p.api.NewTicketProvider()
+	var configs []*model.ConfigWithRows
+	for _, branch := range branches {
+		modelBranch := p.branchesByName[branch.Branch.Name]
 		envPrefix := fmt.Sprintf("TEST_BRANCH_%s_CONFIG", modelBranch.Name)
-		p.createConfigs(pool, branch.Configs, modelBranch, envPrefix)
+		configs = append(configs, p.prepareConfigs(branch.Configs, modelBranch, tickets, envPrefix)...)
+	}
+
+	// Generate new IDs
+	if err := tickets.Resolve(); err != nil {
+		assert.FailNow(p.t, fmt.Sprintf(`cannot generate new IDs: %s`, err))
 	}
+
+	// Add additional ENVs
+	if additionalEnvs != nil {
+		for k, v := range additionalEnvs {
+			p.setEnv(k, testhelper.ReplaceEnvsString(v, p.envs))
+		}
+	}
+
+	// Create requests
+	pool := p.api.NewPool()
+	p.createConfigsRequests(configs, pool)
```

Comment:
A az tak sa vytvoria konfiguracie, ... mozu  sa teda vytvarat paralelne a zaroven na seba odkazovat.
Kedze ID je vygenerovane dopredu, ... 

Opat, najlesie to bude vidiet na priklade v dalsom PR.